File: ulohy/01-fibonacci/fib.py
def fibrek(n):
    return n if n <= 1 else fibrek(n - 1) + fibrek(n - 2)


def fib(n):
    if n <= 1:
        return n
    prev, fib = 0, 1
    for _ in range(2, n + 1):
        fib, prev = fib + prev, fib
    return fib


n = 35
print("Nerekurzivní řešení:\t", fib(n))
print("Rekurzivní řešení:\t", fibrek(n))

File: ulohy/01-fibonacci/fib.php
<?php
function fibrek($n)
{
    return $n <= 1 ? $n : fibrek($n - 1) + fibrek($n - 2);
}

function fib($n)
{
    if ($n <= 1) {
        return $n;
    }
    $prev = 0;
    $fib = 1;
    for ($i = 2; $i <= $n; $i++) {
        $new_fib = $fib + $prev;
        $prev = $fib;
        $fib = $new_fib;
    }
    return $fib;
}

$n = 35;
echo "Nerekurzivní řešení: ", fib($n), "\n";
echo "Rekurzivní řešení  : ", fibrek($n), "\n";


